test/test_specs.py

```diff
@@ -1716,7 +1715,7 @@ def test_stack_binarydiscrete(self, shape, stack_dim):
         c1 = BinaryDiscreteTensorSpec(n=n, shape=shape)
         c2 = c1.clone()
         c = torch.stack([c1, c2], stack_dim)
-        assert isinstance(c, LazyStackedTensorSpec)
+        assert isinstance(c, BinaryDiscreteTensorSpec)
```

Comment:
All of these tests where we no longer return a `LazyStackedTensorSpec` from `torch.stack` may need a second version that tests the lazy stacked behaviour?

Reply:
yep we should probably do that